### DISCUSSION TITLE
CAS or SAML SSO support?

### DISCUSSION THREAD

**Discussion by peg-leg** [Budibase/budibase#3600] [Help] [answered]
Does Budibase support CAS or SAML SSO?

**Comment by mjashanks** [comment, marked as answer]
We support OpenID Connect. 

Looks like CAS supports OpenID: https://www.apereo.org/projects/cas
> Support for multiple protocols (CAS, SAML, OAuth, OpenID)

**Reply by peg-leg** [reply]
Hmm.... thanks for that link.  It's a bit of a confusing read...
How is it that CAS supports CAS and OpenID protocols? If my requirement was to support CAS for authentication, does that mean that OpenID fits the bill, or is it the other way around... if my requirement was to support OpenID, I could use CAS?

![image](https://user-images.githubusercontent.com/20299235/144064727-1af03c1a-0c67-462a-9add-88321bb8c7d4.png)

**Reply by mjashanks** [reply]
It confused me a bit too! I don't know anything about CAS to be honest.

Is CAS a protocol or a project/solution?

If CAS is a solution then it looks like it supports OpenID, therefore it should work with Budibase.

If CAS is a protocol, then we don't support it.

**Reply by peg-leg** [reply]
It looks like there may be a few layers to CAS and the fact that their protocol is also referred to as CAS could be causing confusion.  So I think the Apero CAS server is a centralized authentication system that uses the CAS protocol among others (including OpenID connect), so my take is that if the client has an Apero CAS server in place, it should support OpenID and therefore Budibase...
https://apereo.github.io/cas/4.2.x/protocol/CAS-Protocol.html

*EDIT* this seems to confirm that:
https://stackoverflow.com/questions/43676624/what-is-the-relation-between-cas-vs-saml
